webapp/components/root.jsx

```diff
@@ -93,6 +93,8 @@ export default class Root extends React.Component {
                 browserHistory.push('/signup_user_complete');
             } else if (UserStore.getCurrentUser()) {
                 GlobalActions.redirectUserToDefaultTeam();
+            } else if (Client.getRedirectUrl()) {
+                browserHistory.push('/login?redirect_to=' + Client.getRedirectUrl());
```

Comment:
And if you make that change like in my other comment, this can just redirect to `'/login' + window.location.search`